bindings.c

```diff
@@ -80,6 +80,125 @@ struct file_info {
 	int cached;
 };
 
+/* The function of hash table.*/
+#define LOAD_SIZE 100 /*the size of hash_table */
+static int calc_hash(char *name)
+{
+	unsigned int hash = 0;
+	unsigned int x = 0;
+	/* ELFHash algorithm. */
+	while (*name) {
+		hash = (hash << 4) + *name++;
+		x = hash & 0xf0000000;
+		if (x != 0) {
+			hash ^= (x >> 24);
+			hash &= ~x;
+		}
+	}
+	return ((hash & 0x7fffffff) % LOAD_SIZE);
+}
+
+struct load_node {
+	char *cg;  /*cg */
+	unsigned long avenrun[3];		/* Load averages */
+	unsigned int run_pid;
+	unsigned int total_pid;
+	unsigned int last_pid;
+	int cfd; /* The file descriptor of the mounted cgroup */
+	struct  load_node *next;
+	struct  load_node **pre;
+};
+
+struct load_head {
+	/* The lock is about insert load_node and refresh load_node.To the first load_node of each hash bucket, insert and
```

Comment:
YYY Ok - I'm going to postpone my review of the locking details.
They may or may not be correct, but we can change them later if need be.

In the meantime, though a rationale of the locking would be helpful.

Reply:
Sir，i'm confusd that what you reviewed is not is not the latest.  I rewrite them in a0589c8.
It has five patches totally.

Reply:
Sir，i'm confusd that what you reviewed is not the latest. I rewrite them in a0589c8.
It has five patches totally.

Reply:
Sir，I'm confusd that what you reviewed is not is not the latest. I rewrite them in a0589c8.
It has five patches totally.